### PR TITLE
Add GetMetaStatus / SetMetaStatus to GCS and S3 provider modules

### PR DESCRIPTION
**Pull request by parkerduckworth**

### CODE REVIEW

File: modules/storage-gcs/gcs/gcs.go [hunk 1]
```diff
@@ -208,12 +175,60 @@ func (g *gcs) StoreSnapshot(ctx context.Context, snapshot *snapshots.Snapshot) e
...
 		return errors.Wrap(err, "put file")
 	}
 	return nil
 }
 
+func (g *gcs) GetMetaStatus(ctx context.Context, className, snapshotID string) (string, error) {
+	bucket, err := g.findBucket(ctx)
+	if err != nil {
+		return "", errors.Wrap(err, "get snapshot status")
+	}
```
Comment (aliszka) [inline modules/storage-gcs/gcs/gcs.go:188]: no check if bucket exists?

File: modules/storage-gcs/gcs/gcs.go [hunk 2]
```diff
@@ -208,12 +175,60 @@ func (g *gcs) StoreSnapshot(ctx context.Context, snapshot *snapshots.Snapshot) e
...
+
+	return string(snapshot.Status), nil
+}
+
+func (g *gcs) SetMetaStatus(ctx context.Context, className, snapshotID, status string) error {
+	bucket, err := g.findBucket(ctx)
+	if err != nil {
+		return errors.Wrap(err, "set snapshot status")
+	}
```
Comment (aliszka) [inline modules/storage-gcs/gcs/gcs.go:209]: no check if bucket exists?

File: modules/storage-aws-s3/s3/s3.go [hunk 3]
```diff
@@ -148,6 +148,61 @@ func (s *s3) RestoreSnapshot(ctx context.Context, className, snapshotID string)
...
+	data, err := io.ReadAll(obj)
+	if err != nil {
+		return errors.Wrapf(err, "read file %s", objectName)
+	}
+
+	var snapshot snapshots.Snapshot
+	err = json.Unmarshal(data, &snapshot)
+	if err != nil {
+		return errors.Wrapf(err, "unmarshal meta")
+	}
```
Comment (aliszka) [inline modules/storage-aws-s3/s3/s3.go:188]: that part looks similar to GetMetaStatus -> could be extracted to common func?

File: modules/storage-gcs/gcs/gcs.go (hunk 1 above)
Comment (parkerduckworth) [reply modules/storage-gcs/gcs/gcs.go:188]: fixed

File: modules/storage-gcs/gcs/gcs.go (hunk 2 above)
Comment (parkerduckworth) [reply modules/storage-gcs/gcs/gcs.go:214]: fixed

File: modules/storage-aws-s3/s3/s3.go (hunk 3 above)
Comment (parkerduckworth) [reply modules/storage-aws-s3/s3/s3.go]: done. was able to apply the extracted func to RestoreSnapshot as well

**Pull request merged**